test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -0,0 +1,225 @@
+
+        // Verify that a new reader reads from event 1 onwards.
+        final String newReaderGroupName = readerGroupId + "new";
+        readerGroupManager.createReaderGroup(newReaderGroupName, ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build());
+        @Cleanup
+        final EventStreamReader<String> newReader = clientFactory.createReader(newReaderGroupName + "2",
+                newReaderGroupName, new UTF8StringSerializer(), ReaderConfig.builder().build());
+
+        assertEquals("Expected read event: ", "1", newReader.readNextEvent(5000).getEvent());
+    }
+
+    /**
+    * This method is trying to check the stream seal feature of the stream.
+    * Here we are creating a stream and writing some event to it.
+    * And then sealing that stream by using stream manager and validating the stream whether sealing worked properly.
+    */
+    public  void checkSealStream() {
+        String scopeName = "stream-seal-test-scope";
+        String streamName = "stream-seal-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        assertTrue(streamManager.checkStreamExists(scopeName, streamName));
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
+
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(readerGroupId + "1", readerGroupId,
+                new UTF8StringSerializer(), ReaderConfig.builder().build());
+        // Before sealing of the stream it must write to the existing stream.
+        writer.writeEvent("0").join();
+        assertTrue(streamManager.sealStream(scopeName, streamName));
+        // It must complete Exceptionally because stream is sealed already.
+        assertTrue(writer.writeEvent("1").completeExceptionally(new IllegalStateException()));
+        // But there should not be any problem while reading from a sealed stream.
+        assertEquals(reader.readNextEvent(READER_TIMEOUT_MS).getEvent(), "0");
+    }
+
+    /**
+     * This method is trying to create and delete a scope.
+     * And also validating it.
+     * @throws DeleteScopeFailedException is thrown if this method is unable to seal and delete a stream.
```

Comment:
nit: This method attempts to create, delete and validate the existence of a scope.
@ throws DeleteScopeFailedException if unable to seal and delete a stream